app/src/main/java/com/android/bookswap/data/source/network/BookFirestoreSource.kt

```diff
@@ -27,21 +30,34 @@ class BooksFirestoreSource(private val db: FirebaseFirestore) : BooksRepository
   // Selected todo, i.e the todo for the detail view
   private val selectedBook_ = MutableStateFlow<DataBook?>(null)
   val selectedBook: StateFlow<DataBook?> = selectedBook_.asStateFlow()
-  // Use this code in editBookScreen and modify the editBookScreen structure if needed when
-  // incorporating in the app navigation
 
+  /**
+   * Initializes the Firestore source by adding an authentication state listener. Calls the
+   * onSuccess callback if the user is authenticated.
+   *
+   * @param onSuccess Callback to be invoked when the user is authenticated.
+   */
   override fun init(onSuccess: () -> Unit) {
     Firebase.auth.addAuthStateListener {
       if (it.currentUser != null) {
         onSuccess()
       }
     }
   }
-
+  /**
+   * Generates a new UUID.
+   *
+   * @return a randomly generated UUID
+   */
   override fun getNewUUID(): UUID {
     return UUID.randomUUID()
   }
-
+  /**
+   * Fetches the list of books from the Firestore database.
+   *
+   * @param callback Callback to be invoked with the result of the operation. The result is a list
+   *   of DataBook objects on success, or an exception on failure.
+   */
   override fun getBook(callback: (Result<List<DataBook>>) -> Unit) {
```

Comment:
This method returns a list of book as the comment says but the name `getBook` sggests that it returns a single book. These inconcistencies should also be checked when documenting code.